drivers/char/rust_example.rs

```diff
@@ -86,6 +86,20 @@ impl KernelModule for RustExample {
             mutex_init!(data.as_ref(), "RustExample::init::data1");
             *data.lock() = 10;
             println!("Value: {}", *data.lock());
+
+            // SAFETY: `init` is called below.
+            let cv = Pin::from(Box::try_new(unsafe { CondVar::new() })?);
+            condvar_init!(cv.as_ref(), "RustExample::init::cv1");
+            {
+                let guard = data.lock();
+                #[allow(clippy::while_immutable_condition)]
```

Comment:
Question: will this make the following read on `guard` `volatile`? IOW, could there be anything that stops the compiler from optimizing the following code as

```
let c = *guard;
while c != 10{
    cv.wait(&guard);
}
```

Reply:
I believe the the interior mutability implied by the use of `UnsafeCell` prevents the compiler from assuming that `*guard` won't change: it calls `deref()`, which calls `locked_data()` from the underlying lock, which returns an `UnsafeCell`.